tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -96,7 +158,7 @@ void eth_noc_semaphore_wait(volatile tt_l1_ptr uint32_t* sem_addr, uint32_t val)
 FORCE_INLINE
 void eth_noc_async_read_barrier() {
     while (!ncrisc_noc_reads_flushed(noc_index)) {
```

Comment:
@tt-aho ignore my comments from our conversation yesterday, we can just flush one noc_index